extension/src/types.ts

```diff
@@ -85,3 +85,64 @@ export enum MessageType {
     DEBUG_MESSAGE,
     CHANGE_ROOM,
 }
+
+/**
+ * Video status for one user
+ */
+enum UserVideoStatus {
+    Unknown, // Still Loading of DOM, or not on the video URL, or any othe reason (no info from Video API)
+    Buffering, // has not reached minBufferLength (video is paused or not started)
+    Playing, // is currently playing the video
+    Ready, // (equivalent of Waiting or Paused) has buffered enough of the video, is pending start signal
+}
+
+/**
+ * Video status for all players that have completed initial Sync
+ */
+enum VideoSyncStatus {
+    Paused, // video is paused and no action should be taken (apart from buffering)
+    Waiting, // waiting for all players to be in waiting status before playing
+    Playing, // video should playing normally on all user browsers
+}
+
+/**
+ * Information about one user
+ */
+export class User {
+    id: string; // maybe not needed since it will be the index key
+    connection: string; // The kind of info we need to be able to communicate with the user (TOCONFIRM: just a string is enough ?)
+    joinedAt: number;
+    lastVideoTimestamp: number; // Last video timestamp received during sync events of said user
+    lastHeartbeat: number; // date of last event during sync received from said user
+    currentVideoStatus: UserVideoStatus;
+    initialSync: boolean = false;
+
+    userAgent: string; // Might help debug issues later
+}
+
+interface UserIndex {
+    [id: string]: User;
+}
+
+export class Room {
+    roomId: string;
+    createdAt: number;
+    users: UserIndex;
+    ownerId: string;
+
+    // Config options
+    minBufferLength: number = 5; // Number of seconds each person should have loaded before resuming
+    videoSpeed: number = 1; // speed of video (2 means 2x)
```

Comment:
Do we know how to get/set this value on the html media ?

Reply:
@mourtisma ?